kotsadm/pkg/kurl/kurl_nodes.go

```diff
@@ -0,0 +1,116 @@
+package kurl
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	kuberneteserrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	metrics "k8s.io/metrics/pkg/client/clientset/versioned"
+)
+
+type KurlNodes struct {
+	Nodes         []Node `json:"nodes"`
+	HA            bool   `json:"ha"`
+	IsKurlEnabled bool   `json:"isKurlEnabled"`
+}
+
+type Node struct {
+	Name           string            `json:"name"`
+	IsConnected    bool              `json:"isConnected"`
+	CanDelete      bool              `json:"canDelete"`
+	KubeletVersion string            `json:"kubeletVersion"`
+	CPU            CapacityAvailable `json:"cpu"`
+	Memory         CapacityAvailable `json:"memory"`
+	Pods           CapacityAvailable `json:"pods"`
+	Conditions     NodeConditions    `json:"conditions"`
+}
+
+type CapacityAvailable struct {
+	Capacity  float64 `json:"capacity"`
+	Available float64 `json:"available"`
+}
+
+type NodeConditions struct {
+	MemoryPressure bool `json:"memoryPressure"`
+	DiskPressure   bool `json:"diskPressure"`
+	PidPressure    bool `json:"pidPressure"`
+	Ready          bool `json:"ready"`
+}
+
+// GetNodes will get a list of nodes with stats
+func GetNodes(client kubernetes.Interface, metrics *metrics.Clientset) (KurlNodes, error) {
+	nodes, err := client.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return KurlNodes{}, errors.Wrap(err, "list nodes")
+	}
+
+	toReturn := KurlNodes{}
+
+	for _, node := range nodes.Items {
+		cpuCapacity, _ := strconv.ParseFloat(node.Status.Capacity.Cpu().String(), 64)
```

Comment:
why ignore the errors?

Reply:
because I forgot I hadn't handled them 🤦